pyo3-macros-backend/src/introspection.rs

```diff
@@ -413,3 +462,64 @@ fn ident_to_type(ident: &Ident) -> Cow<'static, Type> {
         .into(),
     )
 }
+
+// Replace Self in types with the given type
+fn replace_self(ty: &mut Type, self_target: &Type) {
```

Comment:
`syn` has a `VisitMut` trait, I wonder if that helps implement this in a way that's more robust?